src/NuGet.Core/NuGet.Packaging/Signing/Specifications/SigningSpecificationsV1.cs

```diff
@@ -9,6 +11,29 @@ public sealed class SigningSpecificationsV1 : SigningSpecifications
         private const string _manifestPath = "testsigned/manifest.txt";
         private const string _signaturePath1 = "testsigned/a.sig";
         private const string _signaturePath2 = "testsigned/b.sig";
+        private const string _SHA256String = "SHA256";
+        private const string _SHA384String = "SHA384";
+        private const string _SHA512String = "SHA512";
+
+        /// <summary>
+        /// Allowed digest algorithms for signature and timestamp hashing.
+        /// </summary>
+        private static readonly string[] _allowedHashAlgorithms = new[]
+        {
+            _SHA256String,
+            _SHA384String,
+            _SHA512String
+        };
+
+        /// <summary>
+        /// Allowed digest algorithm Oids for signature and timestamp hashing.
+        /// </summary>
+        private static readonly string[] _allowedHashAlgorithmOids = new[]
+        {
+            HashAlgorithmName.SHA256.ConvertToOidString(),
+            HashAlgorithmName.SHA384.ConvertToOidString(),
+            HashAlgorithmName.SHA512.ConvertToOidString()
```

Comment:
nit: simplify this if possible since this will be executed as soon as the assembly is loaded